InfrastructureBackupValidator/BackupValidationTool.psm1

```diff
@@ -0,0 +1,591 @@
+        try
+        {
+            $sqlInstance = Get-SqlInstance -ServerInstance $SQLServerInstanceName
+        }
+        catch
+        {
+            throw ($Strings.ErrorFailToConnectSqlServerDefault -f $SQLServerInstanceName, $_)
+        }
+    }
+    else
+    {
+        # STEP 2: Connect to the SQL server with provided credential
+        Write-Verbose ($Strings.ProgressConnectSqlServerProvided -f $SQLServerInstanceName) -Verbose
+        try
+        {
+            $sqlInstance = Get-SqlInstance -ServerInstance $SQLServerInstanceName -Credential $SQLCredential
+            $sqlCommonParams.SQLCredential = $SQLCredential
+        }
+        catch
+        {
+            throw ($Strings.ErrorFailToConnectSqlServerProvided -f $SQLServerInstanceName, $_)
+        }
+    }
+
+    # STEP 3: Create a tmp folder
+    Write-Verbose ($Strings.ProgressCreateTmpFolder -f $SubTmpFolderName, $TempFolder) -Verbose
+    $tmpDir = Join-Path $TempFolder $SubTmpFolderName
+    $null = Remove-Item $tmpDir -Force -Recurse -ErrorAction Ignore | Out-Null
+    $null = New-Item -Type Directory $tmpDir -Force | Out-Null
+    
+    # STEP 4: Get the list of backup files according to the BackupID
+    Write-Verbose ($Strings.ProgressGetBackupSnapshots -f $BackupID) -Verbose
+    $backupRoot = Join-Path $BackupStorePath "MASBackup\progressivebackup"
+    $backupFiles = (Get-ChildItem -Path $backupRoot -Recurse | ? { $_.Name -match $BackupID -and $_.Name -match ".zip" }).FullName
+    $backupRepos = @($BackupRepoNames.Values.GetEnumerator()) | ? { $_ -ne $BackupRepoNames.Subscription }
+    $backupFiles = $backupFiles | ? { $backupRepos.Contains(($_ -Split "\\")[-2]) }
+    if ($null -eq $backupFiles -or $backupFiles.Count -eq 0)
+    {
+        throw ($Strings.ErrorFailToFindBackupSnapshots -f ($backupRepos -join ","))
+    }
+
+    $diff = Compare-Object -ReferenceObject $backupRepos -DifferenceObject ($backupFiles | % { ($_ -Split "\\")[-2] })
+    $missingRepos = ($diff | ? { $_.SideIndicator -eq "<=" }).InputObject
+    if ($null -ne $missingRepos -and $missingRepos.Count -gt 0)
+    {
+        throw ($Strings.ErrorFailToFindBackupSnapshots -f ($missingRepos -join ","))
+    }
+
+    $duplicateRepos = ($diff | ? { $_.SideIndicator -eq "=>" }).InputObject
+    if ($null -ne $duplicateRepos -and $duplicateRepos.Count -gt 0)
+    {
+        throw ($Strings.ErrorFindMoreThanOneSnapshot -f ($duplicateRepos -join ","))
+    }
+
+    # It's progressive backup for Subscription, get the backup chain
+    $backupStoreName = ($backupFiles[0] -Split "\\")[-3]
+    $subscriptionBackupChain = Get-RepositoryBackupChain -ExternalShare $BackupStorePath `
+        -ExternalShareCredential $BackupStoreCredential -ProgressiveBackupStoreName $backupStoreName `
+        -RepositoryName $BackupRepoNames.Subscription -BackupId $BackupID
+    $subscriptionBackups = @()
+    foreach ($backup in $subscriptionBackupChain)
+    {
+        $subscriptionBackups += $backup.DataFileName
```

Comment:
The above cmdlet already takes a backup Id, doesn't it return a chain up-to and including that particular backup Id without any possible future backups?